### PR TITLE
source-marketstack contribution from ombhardwajj

### LINKED ISSUE #47389: [connector-request] Marketstack
This is a suggestion to create a connector for Marketstack using the UI Connector Builder.
The website for the connector is https://marketstack.com/

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to step 3).
3. Describe steps to get credentials.
4. Create the connector using the UI Builder.   

> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> This is a suggestion, and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend following the steps above.

### PR DESCRIPTION
**Pull request by ombhardwajj**
This closes #47389
## What

This PR adds a new connector for source Marketstack (source-marketstack).

Contributor-provided connector description:

Marketstack provides data from 72 global stock exchanges.
Using this connector we can extract Historical Data , Splits and Dividends data !
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |
| `from` | `string` | From.  |  |
| `to` | `string` | To.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| Exchanges | mic | DefaultPaginator | ✅ |  ❌  |
| Tickers | symbol | DefaultPaginator | ✅ |  ❌  |
| Historical Data |  | DefaultPaginator | ✅ |  ✅  |
| Splits |  | DefaultPaginator | ✅ |  ❌  |
| Dividends |  | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by marcosmarxm** [comment]
All contributions must be merged by Nov-14. I'm going to work to provide a first review by Friday to give users the weekend to fix any comment. Please be aware of this.

Make sure you have pagination properly working, the API doesn't have pagination please left a comment making my review easier.
Make sure have a square svg icon
Make sure credentials are valid.

File: airbyte-integrations/connectors/source-marketstack/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+            - type: ParentStreamConfig
+              parent_key: mic
+              partition_field: mic
+              stream:
+                $ref: "#/definitions/streams/Exchanges"
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/Tickers"
+    Historical Data:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-marketstack/manifest.yaml:97]: This stream must be incremental. Please fix that.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left one comment, than ready to go.

**Comment by marcosmarxm** [comment]
Molving next check to Nov 11 Monday

File: airbyte-integrations/connectors/source-marketstack/manifest.yaml (hunk 1 above)
Comment (ombhardwajj) [reply airbyte-integrations/connectors/source-marketstack/manifest.yaml:97]: done!!

**Comment by ombhardwajj** [comment]
Historical Data now supports incremental sync!!
@marcosmarxm

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**